src/main/java/software/amazon/encryption/s3/internal/ReEncryptInstructionFileRequest.java

```diff
@@ -0,0 +1,154 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package software.amazon.encryption.s3.internal;
+
+import software.amazon.encryption.s3.S3EncryptionClientException;
+import software.amazon.encryption.s3.materials.AesKeyring;
+import software.amazon.encryption.s3.materials.RawKeyring;
+import software.amazon.encryption.s3.materials.RsaKeyring;
+
+/**
+ * Request object for re-encrypting instruction files in S3.
+ * This request supports re-encryption operations using either AES or RSA keyrings.
+ * For AES keyrings, only the default instruction file suffix is supported.
+ * For RSA keyrings, a custom instruction file suffix must be provided to support
+ * multiple accesses to the same encrypted object.
+ */
+public class ReEncryptInstructionFileRequest {
+  private final String bucket;
+  private final String key;
+  private final RawKeyring newKeyring;
+  private final String instructionFileSuffix;
+
+  private ReEncryptInstructionFileRequest(Builder builder) {
+    bucket = builder.bucket;
+    key = builder.key;
+    newKeyring = builder.newKeyring;
+    instructionFileSuffix = builder.instructionFileSuffix;
+  }
+
+  /**
+   * @return the S3 bucket name that contains the encrypted object and instruction file to re-encrypt
+   */
+  public String bucket() {
+    return bucket;
+  }
+
+  /**
+   * @return the S3 object key of the encrypted object whose instruction file will be re-encrypted
+   */
+  public String key() {
+    return key;
+  }
+
+  /**
+   * @return the new keyring (AES or RSA) that will be used to re-encrypt the instruction file
+   */
+  public RawKeyring newKeyring() {
+    return newKeyring;
+  }
+
+  /**
+   * @return the suffix to use for the instruction file. The default instruction file suffix is ".instruction" for
+   * AES keyrings and the instruction file suffix must be different from the default one for RSA keyrings
+   */
+  public String instructionFileSuffix() {
+    return instructionFileSuffix;
+  }
+
+  /**
+   * Creates a builder that can be used to configure and create a {@link ReEncryptInstructionFileRequest}
+   *
+   * @return a new builder
+   */
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * Builder for ReEncryptInstructionFileRequest.
+   */
+  public static class Builder {
+    private static final String DEFAULT_INSTRUCTION_FILE_SUFFIX = ".instruction";
```

Comment:
This definition must not be duplicated. Use the existing `INSTRUCTION_FILE_SUFFIX` (though we should rename that one).